### ISSUE TITLE
Locking Terraform version 

### ISSUE THREAD

**Issue by Kausik-A** [aws-observability/aws-otel-collector-test-framework#608]
The current terraform scripts run on the latest releases provided by Hashcorp.These tf scripts are a integral part of the testing workflows under OTEL.So it is generally a good practice to lock down the versions of the terraform script to improve stability.

**Comment by Kausik-A** [comment]
hey @anuraaga, need your advice if locking the terraform state files is a good idea or its okay to continue use the latest version.

**Comment by anuraaga** [comment]
@Kausik-A Yeah it's generally a good idea to lock the version of terraform in CI, and to use a terraform lock file to lock the versions of used providers / dependencies.
